server/service/src/sensor/berlinger.rs

```diff
@@ -0,0 +1,380 @@
+            )
+        }
+        BreachType::HotConsecutive => {
+            config_description = format!(
+                "Consecutive {} hotter than {}",
+                config_description, temperature_breach_config.maximum_temperature
+            )
+        }
+        BreachType::HotCumulative => {
+            config_description = format!(
+                "Cumulative {} hotter than {}",
+                config_description, temperature_breach_config.maximum_temperature
+            )
+        }
+    }
+
+    let result =
+        get_matching_sensor_breach_config(connection, &config_description, &breach_row_type)?;
+
+    if let Some(_record) = result.clone().pop() {
+        Ok(())
+    } else {
+        let new_temperature_breach_config = TemperatureBreachConfigRow {
+            id: uuid(),
+            store_id: sensor_row.store_id.clone(),
+            is_active: true,
+            description: config_description.clone(),
+            duration: temperature_breach_config.duration.num_seconds() as i32,
+            r#type: breach_row_type,
+            minimum_temperature: temperature_breach_config.minimum_temperature,
+            maximum_temperature: temperature_breach_config.maximum_temperature,
+        };
+        TemperatureBreachConfigRowRepository::new(connection)
+            .upsert_one(&new_temperature_breach_config)?;
+        println!(
+            "Added sensor breach config {:?} ",
+            new_temperature_breach_config
+        );
+        Ok(())
+    }
+}
+
+fn sensor_add_if_new(
+    connection: &StorageConnection,
+    store_id: &str,
+    temperature_sensor: &temperature_sensor::Sensor,
+) -> Result<(), RepositoryError> {
+    let result = get_matching_sensor_serial(connection, &temperature_sensor.serial)?;
+
+    if let Some(_record) = result.clone().pop() {
+        Ok(())
+    } else {
+        let mut interval_seconds = None;
+        if let Some(interval_duration) = temperature_sensor.log_interval {
+            interval_seconds = Some(interval_duration.num_seconds() as i32);
+        }
+        let new_sensor = SensorRow {
+            id: uuid(),
+            serial: temperature_sensor.serial.clone(),
+            name: temperature_sensor.name.clone(),
+            store_id: Some(store_id.to_string()),
+            location_id: None,
+            last_connection_datetime: None,
+            battery_level: None,
+            is_active: true,
+            log_interval: interval_seconds,
+            r#type: SensorType::Berlinger,
+        };
+        SensorRowRepository::new(connection).upsert_one(&new_sensor)?;
+        println!("Added sensor {:?} ", new_sensor);
+        Ok(())
+    }
+}
+
+pub fn read_sensors(
+    connection: &StorageConnection,
+    store_id: &str,
+) -> Result<Vec<String>, SingleRecordError> {
+    let mut sensors_processed: Vec<String> = Vec::new();
+
+    let sensor_serials =
+        read_connected_serials().map_err(|err| SingleRecordError::NotFound(err))?;
```

Comment:
As per earlier comment in issues, I think client or web app should have `upload` option for now, and we just use that uploaded file (since users would expect to upload logs from client machine, not server machine). And we can add native Electron and Capacitor (Electron) functionality to find sensor files through USB in the future

Reply:
Yes, I've added `parse_sensor()` to the `temperature-sensor` crate now so that it can take the contents of an uploaded file and send it to the server for processing. That'll need a bit of a re-jig to take the common code out of `read_sensors` and into a fn which can be called for both USB and direct file contents...